Outbox: Information regarding accessing outboxed services (#725)
cap-js/docs: java/outbox.md

A service wrapped by an outbox can be unboxed by calling the API `OutboxService.unboxed(Service)`. Method calls to the unboxed
service are executed synchronously without storing the event in an outbox.

::: warning Java Proxy

> A service wrapped by an outbox is a [Java Proxy](https://docs.oracle.com/javase/8/docs/technotes/guides/reflection/proxy.html). Such a proxy only implements the _interfaces_ of the object they are wrapping. This means an outboxed service proxy can't be casted to the class implementing the underlying service object.

Reviewer: ```suggestion
A service wrapped by an outbox is a [Java Proxy](https://docs.oracle.com/javase/8/docs/technotes/guides/reflection/proxy.html). Such a proxy only implements the _interfaces_ of the object it is wrapping. This means an outboxed service proxy can't be casted to the class implementing the underlying service object.
```

A service wrapped by an outbox can be unboxed by calling the API `OutboxService.unboxed(Service)`. Method calls to the unboxed
service are executed synchronously without storing the event in an outbox.

::: warning Java Proxy

> A service wrapped by an outbox is a [Java Proxy](https://docs.oracle.com/javase/8/docs/technotes/guides/reflection/proxy.html) that
> that only provides access to the service instance via the interfaces it implements. It can't be casted to a class implementing
> the specific service. All messaging services and the auditlog services are automatically provided as services wrapped by an outbox.

Reviewer: ```suggestion
A service wrapped by an outbox is a [Java Proxy](https://docs.oracle.com/javase/8/docs/technotes/guides/reflection/proxy.html). Such a proxy only implements the _interfaces_ of the service they are wrapping. This means an outboxed service proxy can't be casted to the class implementing the underlying service object.
```

Examples for common service interfaces:

- [com.sap.cds.services.auditlog.AuditLogService](https://javadoc.io/static/com.sap.cds/cds-services-api/1.23.1/com/sap/cds/services/auditlog/AuditLogService.html)
- [com.sap.cds.services.messaging.MessagingService](https://javadoc.io/static/com.sap.cds/cds-services-api/1.23.1/com/sap/cds/services/messaging/MessagingService.html)
- [com.sap.cds.services.cds.CqnService](https://javadoc.io/static/com.sap.cds/cds-services-api/1.23.1/com/sap/cds/services/cds/CqnService.html)

> - [com.sap.cds.services.cds.RemoteService](https://javadoc.io/static/com.sap.cds/cds-services-api/1.23.1/com/sap/cds/services/cds/RemoteService.html)

Reviewer: I am not sure if we need these examples.
